Dockerfile

```diff
@@ -1,15 +1,13 @@
 FROM golang:1.4
 
 ENV CONFIG_PATH /etc/docker/registry/config.yml
-RUN mkdir -pv "$(dirname $CONFIG_PATH)"
-
 ENV DISTRIBUTION_DIR /go/src/github.com/docker/distribution
+ENV GOPATH $DISTRIBUTION_DIR/Godeps/_workspace:$GOPATH
+
 WORKDIR $DISTRIBUTION_DIR
 COPY . $DISTRIBUTION_DIR
-ENV GOPATH $GOPATH:$DISTRIBUTION_DIR/Godeps/_workspace
-
-RUN go install -v ./cmd/registry
-
+RUN make PREFIX=/go clean binaries
+RUN mkdir -pv "$(dirname $CONFIG_PATH)"
```

Comment:
That likely breaks if config path contains a space (yeah, spaces in paths is calling for trouble but then...).